packages/omgidl-serialization/src/MessageReader.ts

```diff
@@ -262,16 +207,40 @@ export class MessageReader<T = unknown> {
         } else {
           return deser(reader, arrayLengths[0]!);
         }
+      } else if (field.typeDeserInfo.type === "primitive") {
+        return field.typeDeserInfo.deserialize(
+          reader,
+          headerSpecifiedLength, // fieldLength only used for `string` type primitives
+        );
       } else {
-        const deser = deserializers.get(field.type);
-        if (deser == undefined) {
-          throw new Error(`Unrecognized primitive type ${field.type}`);
-        }
+        throw new Error(`Unhandled deserialization info type`);
+      }
+    }
+  }
 
-        // fieldLength only used for `string` type primitives
-        return deser(reader, headerSpecifiedLength);
+  private readComplexNestedArray(
```

Comment:
is `readNestedArray` necessary anymore?

Reply:
Yes, this is used for reading typed arrays